### PR TITLE
limchengyang/add_smartystreet_address_check

### PR DESCRIPTION
**Pull request by chengyang-binary**

### CODE REVIEW

File: Changes [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Revision history for {{$dist->name}}
+
+{{$NEXT}}
+
+        Pre-release version, Released as an proof of concept.
```
Comment (tom-binary) [inline Changes]:
```suggestion
        Pre-release version. Released as a proof of concept.
```

File: lib/WebService/Async/SmartyStreets.pm [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+our $VERSION = '0.001';
+
+=HEAD
+
+WebService::Async::SmartyStreets - calls the SmartyStreets API and checks for the validity of the address
+
+=head1 SYNOPSIS
+    
+    my $ss;
```
Comment (tom-binary) [inline lib/WebService/Async/SmartyStreets.pm]: No need for a separate line, just declare + define in one statement.

File: lib/WebService/Async/SmartyStreets.pm [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+    $ss = WebService::Async::SmartyStreets->new(
+        auth_id => #insert auth_id,
+        token   => #insert token,
+        );
+    IO::Async::Loop->new->add($ss);
+
+    $ss = get_smartystreets();
```
Comment (tom-binary) [inline lib/WebService/Async/SmartyStreets.pm]: This needs to be removed.

File: lib/WebService/Async/SmartyStreets.pm [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+use JSON::MaybeUTF8 qw(:v1);
+
+use WebService::Async::SmartyStreets::Address;
+
+use Log::Any qw($log);
+
+=head2 configure
+
+Configures the class with the auth_id and token
+Takes in: Hash which consists of auth_id and token
```
Comment (tom-binary) [inline lib/WebService/Async/SmartyStreets.pm]: This is misleading and the formatting makes it hard to see the full list. Please see our official documentation for this.

```
Takes the following named parameters:

=over 4

=item * C<auth_id> - ...

=item * C<token> - ...

=back

```

File: lib/WebService/Async/SmartyStreets.pm [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+            ));
+        $ua;
+        }
+}
+
+=head2 verify
+
+Makes connection to SmartyStreets API and parses the response into WebService::Async::SmartyStreets::Address.
+
+    my $ss = WebService::Async::SmartyStreets::International->new(
```
Comment (tom-binary) [inline lib/WebService/Async/SmartyStreets.pm]: None of this is needed - it's the same as any other method documentation: just show how to call **this method** and handle the result, you can assume that there's already an instance.

Comment (chengyang-binary) [reply lib/WebService/Async/SmartyStreets.pm]: alright

File: lib/WebService/Async/SmartyStreets.pm [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+        auth_id => "...",
+        token   => "...",
+    );
+    IO::Async::Loop->new->add($ss);
+
+    my $addr = $ss->verify("URI String", %address_to_check)->get;
```
Comment (tom-binary) [inline lib/WebService/Async/SmartyStreets.pm]: For an end-user, what does "URI String" mean here?

This method is not really supposed to be called externally, the URI is something that the `_international`/`_usa` methods will set, so if we did want to encourage usage of this method we'd have to point to the relevant parts of the official documentation for the URI parameter, or at least suggest looking at the other two methods.

File: lib/WebService/Async/SmartyStreets.pm [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+
+Takes the following named parameters:
+
+=over 4
+
+=item * C<uri> - URI address (in string)
```
Comment (tom-binary) [inline lib/WebService/Async/SmartyStreets.pm]: POD is a paragraph-based system, all `=directives` need to have a blank line before + after. See http://perldoc.pl/perlpod for more details.

File: lib/WebService/Async/SmartyStreets.pm [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+=item * C<args> - address parameters in hash (See L<WebService::Async::SmartyStreets/verify_international>)
```
Comment (tom-binary) [inline lib/WebService/Async/SmartyStreets.pm]: It's not a hash when you pass it to a function - it's just a list of keys and values. Calling it a hash is misleading and can lead people to think that they should provide a hashref.

File: lib/WebService/Async/SmartyStreets.pm [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+
+Takes the following named parameters:
+
+=over 4
+
+=item * C<uri> - URI address that the process will make the call to
+
+=back 
+
+More information of the resposne can be seen in L<SmartyStreets Documentation | https://smartystreets.com/docs/cloud/international-street-api>
```
Comment (tom-binary) [inline lib/WebService/Async/SmartyStreets.pm]:
```suggestion
More information on the response can be seen in L<SmartyStreets Documentation | https://smartystreets.com/docs/cloud/international-street-api>
```

File: lib/WebService/Async/SmartyStreets.pm [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+=head2 get_uri
+
+Dummy sub designed to be overriden in L<WebService::Async::SmartyStreets::International> and L<WebService::Async::SmartyStreets::USA>
+
+=cut
+
+sub get_uri {
+    die "Subroutine not overriden in the child's module";
```
Comment (tom-binary) [inline lib/WebService/Async/SmartyStreets.pm]: Neither of those modules are even mentioned in the description.

File: lib/WebService/Async/SmartyStreets.pm [hunk 11]
```diff
@@ -0,0 +1,210 @@
+package WebService::Async::SmartyStreets;
+# ABSTRACT: Access SmartyStreet API
+
+use strict;
+use warnings;
+
+our $VERSION = '0.001';
+
+=HEAD
```
Comment (tom-binary) [inline lib/WebService/Async/SmartyStreets.pm]: invalid POD

```suggestion
=head1 NAME
```

**Pull request merged**